contracts/ERC721B.sol

```diff
@@ -254,9 +252,7 @@ abstract contract ERC721B {
         address to,
         uint256 id
     ) public virtual {
-        transferFrom(from, to, id);
-
-        if (!_checkOnERC721Received(from, to, id, '')) revert TransferToNonERC721ReceiverImplementer();
+        safeTransferFrom(from, to, id, '');
```

Comment:
Since 0.8.4, `safeTransferFrom(address, address, uint, bytes)` would be inlined here, no need to duplicate the logic, you can simply calling the function.